lib/ansible/modules/storage/purestorage/purefb_fs.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Simon Dodsley (simon@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: purefb_fs
+version_added: "2.5"
+short_description:  Manage filesystemon Pure Storage FlashBlade`
+description:
+    - This module manages filesystems on Pure Storage FlashBlade.
+author: Simon Dodsley (@simondodsley)
+options:
+  name:
+    description:
+      - Filesystem Name.
+    required: true
+  state:
+    description:
+      - Create, delete or modifies a filesystem.
+    required: false
+    default: present
+    choices: [ "present", "absent" ]
+  eradicate:
+    description:
+      - Define whether to eradicate the filesystem on delete or leave in trash.
+    required: false
+    type: bool
+    default: false
+  size:
+    description:
+      - Volume size in M, G, T or P units. See examples.
+    required: false
+    default: 32G
+  nfs:
+    description:
+      - Define whether to NFS protocol is enabled for the filesystem.
+    required: false
+    type: bool
+    default: true
+  nfs_rules:
+    description:
+      - Define the NFS rules in operation.
+    required: false
+    type: bool
+    default: '*(rw,no_root_squash)'
+  smb:
+    description:
+      - Define whether to SMB protocol is enabled for the filesystem.
+      - Define whether to SMB protocol is enabled for the filesystem.
+    required: false
+    type: bool
+    default: false
+  http:
+    description:
+      - Define whether to HTTP/HTTPS protocol is enabled for the filesystem.
```

Comment:
Doesn't sound correct.
`Define whether HTTP/HTTPS protocol should be enabled for the filesystem.`?

Reply:
@gundalow The device I'm doing this module for can have HTTP or HTTPS access to the filesystem as well as classic NFS/SMB. Newer applications can use these protocols to stream files